internal/pkg/manifest/svc.go

```diff
@@ -664,32 +664,45 @@ func prepareParsedExposedPortsMap(exposedPorts []ExposedPort) (map[string][]Expo
 
 // NetworkLoadBalancerTarget returns target container and target port for the NLB configuration.
 // This method should be called only when NLB config is not empty.
-func (s *LoadBalancedWebService) NetworkLoadBalancerTarget() (targetContainer string, targetPort string, err error) {
-	// Configure target container and port.
-	targetContainer = aws.StringValue(s.Name)
-	if s.NLBConfig.Listener.TargetContainer != nil {
-		targetContainer = aws.StringValue(s.NLBConfig.Listener.TargetContainer)
+func (s *LoadBalancedWebService) NetworkLoadBalancerTarget(lContainer string, lPort *int, nlbPort string) (targetContainer string, targetPort string, err error) {
```

Comment:
We could have it but there are multiple different things that we need here to make a final decision about `targetContainer` and `targetPort`.  Firstly we need  `exposedPorts` that we are planning to pass as a method input. secondly, we need `workload` because we assign it as default `targetContainer`. Thirdly, we need `s.Sidecars` in case we only have `targetContainer` mentioned and if it is sidecar container then we need to pull port of the sidecar container to set it as `targetPort`. So if we decide to set this method at listener level then we will anyway need to send these parameters as input to this method. Do you have any thoughts on this?